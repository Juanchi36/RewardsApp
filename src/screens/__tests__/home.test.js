import React from 'react';
import { render } from '@testing-library/react-native';
import { Home } from '@screens';
import { MockContainer } from '@helpers';

describe('home', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should return a snapshot of the screen', () => {
    const result = render(
      <MockContainer>
        <Home />
      </MockContainer>
    );

    const screen = result.getByTestId('home-screen');

    expect(screen).toBeDefined();
    expect(result.toJSON()).toMatchSnapshot();
  });
});
