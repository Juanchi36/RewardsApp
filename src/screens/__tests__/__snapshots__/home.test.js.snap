// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home should return a snapshot of the screen 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="MockedScreen"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTSafeAreaView
            style={
              [
                {
                  "backgroundColor": "#f6f6f6",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
            testID="home-screen"
          >
            <View
              style={
                [
                  {
                    "marginTop": 20,
                    "paddingHorizontal": 20,
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#020202",
                      "fontFamily": "Avenir-Medium",
                      "fontSize": 20,
                      "fontWeight": "800",
                      "lineHeight": 27.32,
                    },
                  ]
                }
              >
                Bienvenido de vuelta!
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#020202",
                      "fontFamily": "Avenir-Medium",
                      "fontSize": 14,
                      "fontWeight": "400",
                      "lineHeight": 19.12,
                    },
                  ]
                }
              >
                Ruben Rodriguez
              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#9b9898",
                      "fontFamily": "Avenir-Medium",
                      "fontSize": 14,
                      "fontWeight": "800",
                      "lineHeight": 19.12,
                      "paddingVertical": 20,
                    },
                  ]
                }
              >
                TUS PUNTOS
              </Text>
              <View
                style={
                  [
                    {
                      "paddingHorizontal": "8%",
                    },
                  ]
                }
              >
                <View
                  style={
                    [
                      {
                        "backgroundColor": "#334ffa",
                        "borderRadius": 20,
                        "elevation": 1,
                        "height": 143,
                        "justifyContent": "center",
                        "shadowOffset": {
                          "height": 4,
                          "width": 0,
                        },
                        "shadowOpacity": 0.5,
                        "shadowRadius": 20,
                      },
                    ]
                  }
                  testID="points-summaty-container"
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#ffffff",
                          "fontFamily": "Avenir-Medium",
                          "fontSize": 16,
                          "fontWeight": "800",
                          "left": 18,
                          "lineHeight": 21.86,
                          "position": "absolute",
                          "top": 21,
                        },
                      ]
                    }
                  >
                    Diciembre
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "color": "#ffffff",
                          "fontFamily": "Avenir-Medium",
                          "fontSize": 32,
                          "fontWeight": "800",
                          "lineHeight": 43.71,
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    0.00 pts
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                [
                  {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "paddingHorizontal": 20,
                    "paddingVertical": 20,
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#9b9898",
                      "fontFamily": "Avenir-Medium",
                      "fontSize": 14,
                      "fontWeight": "800",
                      "lineHeight": 19.12,
                    },
                  ]
                }
              >
                TUS MOVIMIENTOS
              </Text>
              <View
                smallDevice={false}
                style={
                  [
                    {
                      "backgroundColor": "#fff",
                      "borderRadius": 10,
                      "marginTop": 20,
                      "maxHeight": "77%",
                      "paddingHorizontal": 15,
                      "paddingVertical": 20,
                    },
                  ]
                }
              />
            </View>
            <View
              style={
                [
                  {
                    "bottom": "4%",
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                    "left": "0%",
                    "paddingHorizontal": 20,
                    "position": "absolute",
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#334ffa",
                    "borderRadius": 10,
                    "height": 50,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": "48%",
                  }
                }
                testID="primary-button"
              >
                <Text
                  style={
                    [
                      {
                        "color": "#ffffff",
                        "fontFamily": "Avenir-Medium",
                        "fontSize": 16,
                        "fontWeight": "800",
                        "lineHeight": 17.39,
                      },
                    ]
                  }
                >
                  Ganados
                </Text>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#334ffa",
                    "borderRadius": 10,
                    "height": 50,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": "48%",
                  }
                }
                testID="primary-button"
              >
                <Text
                  style={
                    [
                      {
                        "color": "#ffffff",
                        "fontFamily": "Avenir-Medium",
                        "fontSize": 16,
                        "fontWeight": "800",
                        "lineHeight": 17.39,
                      },
                    ]
                  }
                >
                  Canjeados
                </Text>
              </View>
            </View>
          </RCTSafeAreaView>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
